ntp-proto/src/nts_record.rs

```diff
@@ -1504,7 +1522,17 @@ impl KeyExchangeServer {
                         ControlFlow::Continue(decoder) => {
                             // more bytes are needed
                             self.state = State::Active { decoder };
-                            ControlFlow::Continue(self)
+
+                            // recursively invoke the progress function. This is very unlikely!
+                            //
+                            // Normally, all records are written with a single write call, and
+                            // received as one unit. Using many write calls does not really make
+                            // sense for a client.
+                            //
+                            // So then, the other reason we could end up here is if the buffer is
+                            // full. But 512 bytes is a lot of space for this interaction, and
+                            // should be sufficient in most cases.
+                            self.progress()
```

Comment:
Two questions:
- What's the advantage over a recursive call vs. using the `ControlFlow::Continue(self)`?
- How would this behave if someone (i.e. an attacker) starts making it very likely that this happens with the objective of making the NTS server crash due to a stack overflow?